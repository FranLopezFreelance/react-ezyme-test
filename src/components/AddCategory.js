import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({setCategory}) => {
  
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (e) => {
    setInputValue(e.target.value)
  }

  const handleAddCategory = () => {
    if(inputValue.trim().length > 1){
      setCategory(cats => [inputValue, ...cats ]);
      setInputValue('');
    }
  }

  const handleSubmit = (e) => {
    e.preventDefault();
    if(inputValue.trim().length > 1){
      setCategory(cats => [inputValue, ...cats ]);
      setInputValue('');
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <input 
        placeholder="Buscar... "
        type="text" 
        className="category-input" 
        value={inputValue} 
        onChange={handleInputChange} />
      <button 
        className="add-category"
        onClick={handleAddCategory}>Buscar</button>
    </form>
  )
}

AddCategory.propTypes = {
  setCategory: PropTypes.func.isRequired
}

export default AddCategory
