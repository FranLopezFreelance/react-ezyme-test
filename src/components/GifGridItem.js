import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ({url, title}) => {
  return (
    <div className="gif">
      <p>{title}</p>
      <img className="gif-img" src={url} alt={title} />
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}

export default GifGridItem;
