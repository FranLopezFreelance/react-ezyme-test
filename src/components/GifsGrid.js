import React, { useState, useEffect, Fragment } from 'react';
import PropTypes from 'prop-types';
import { GifGridItem } from './GifGridItem';

export const GifsGrid = ({category}) => {

  const [images, setImages] = useState([]);
  useEffect(() => {
    getGifs();
  }, []);

  const apiKey = 'eU0RKJE15pCeTK4baaf03vzbKUMso0x3';
  const url = `https://api.giphy.com/v1/gifs/search?q=${encodeURI(category)}&limit=12&api_key=${apiKey}`;
  const getGifs = async () => {
    const res = await fetch(url);
    const {data} = await res.json();
    let gifs = [];
    if(data){
      gifs = data.map(({id, title, images}) => {
        const {downsized_medium: url} = images;
        return {id, title, url: url.url};
      });
    }
    
    setImages(gifs);
  }

  return (
    <Fragment>
      <h3>{category}</h3>
        {
          images.map(image => {
            return <GifGridItem 
                      key={image.id} 
                      {...image} />
          })
        }
    </Fragment>
  )
}

GifsGrid.propTypes = {
  category: PropTypes.string.isRequired
}

export default GifsGrid;
