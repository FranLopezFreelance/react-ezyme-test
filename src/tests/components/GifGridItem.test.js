import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {

  const title = 'Título';
  const url = 'https://url.com';
  const wrapper = shallow(<GifGridItem url={url} title={title} />);

  test('debe mostrar el componente correctamente', () => {
    expect( wrapper ).toMatchSnapshot();
  });

  test('p debe tener el título', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });

  test('img debe tener url en src', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url);
  });

  test('a debe tener className animate__animated', () => {
    const a = wrapper.find('a');
    const className = a.prop('className');
    expect(className.includes('animate__animated')).toBe(true);
  });
});
