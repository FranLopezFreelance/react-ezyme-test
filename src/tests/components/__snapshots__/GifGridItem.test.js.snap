// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe mostrar el componente correctamente 1`] = `
<a
  className="animate__animated animate__backInUp gif"
  href="https://url.com"
  target="blank"
  title="Título"
>
  <p>
    Título
  </p>
  <img
    alt="Título"
    className="gif-img"
    src="https://url.com"
  />
</a>
`;
